webapp/boards/src/components/shareBoard/shareBoardLoginButton.tsx

```diff
@@ -12,14 +12,17 @@ import './shareBoardLoginButton.scss'
 
 const ShareBoardLoginButton = () => {
     const match = useRouteMatch<{teamId: string, boardId: string, viewId?: string, cardId?: string}>()
-    const history = useHistory()
 
-    const redirectQueryParam = 'r=' + encodeURIComponent(generatePath('/:boardId?/:viewId?/:cardId?', match.params))
-    const loginPath = '/login?' + redirectQueryParam
+    // Mattermost login doesn't respect the redirect query parameter
+    // if the user is already logged in, so we send the user to the
+    // board and if they are not logged in, the webapp will take care
+    // of the redirection
```

Comment:
> Mattermost login doesn't respect the redirect query parameter if the user is already logged in

Seems like something that should be fixed so that redirect is handled for logged in users

Reply:
The comment was more useful before mono-repo when boards had different build targets (as in standalone versions) that behave differently. I think that the current behavior (direct the user to the final URL and let the platform handle the login flow if necessary) is correct, but I've created https://mattermost.atlassian.net/browse/MM-51876 in case we want to handle the redirections for logged in users differently.